ethcontract/src/transport.rs

```diff
@@ -62,10 +73,11 @@ pub struct DynTransport {
 
 impl DynTransport {
     /// Wrap a `Transport` in a `DynTransport`
-    pub fn new<F, T>(inner: T) -> Self
+    pub fn new<F, B, T>(inner: T) -> Self
     where
         F: Future<Output = Result<Value, Web3Error>> + Send + 'static,
-        T: Transport<Out = F> + Sync + Send + 'static,
+        B: Future<Output = Result<Vec<Result<Value, Web3Error>>, Web3Error>> + Send + 'static,
+        T: Transport<Out = F> + BatchTransport<Batch = B> + Sync + Send + 'static,
```

Comment:
The way this is written (and in the generated code before) means that the transport has to be Transport AND BatchTransport whereas before only Transport was required. Maybe this isn't a problem because in practice all Transports are also BatchTransport but I wonder if there is a way to not require this.
Maybe there could be a DynBatchTransport which is like DynTransport but with Batch?
Or DynTransport could only implement BatchTransport if the inner Transport is BatchTransport.

Reply:
> Maybe there could be a DynBatchTransport which is like DynTransport but with Batch?

I was thinking about it, see [above][1].

> Or DynTransport could only implement BatchTransport if the inner Transport is BatchTransport.

This would be the best, but it's not possible because `DynTransport` is not generic.

[1]: https://github.com/gnosis/ethcontract-rs/pull/601#issuecomment-900183894

Reply:
If you think that `DynBatchTransport` should be included in this PR, I can implement it. Maybe someone else have something to add?

Reply:
Didn't see your message when I was reviewing. What you say makes sense. It would be nice if there was DynBatchTransport but I don't think it is necessary. Like I said, in practice all transports do implement it. I don't think it is a big deal to breaking change this later either since this is a fairly niche use case.